src/features/filter.ts

```diff
@@ -2,11 +1,25 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Status } from '../types/Status';
 
-const initialState = {
+type Filter = {
+  query: string;
+  status: Status;
+};
+
+const initialState: Filter = {
   query: '',
   status: 'all',
 };
 
 export const filterSlice = createSlice({
   name: 'filter',
   initialState,
-  reducers: {},
+  reducers: {
+    setStatus(filter: Filter, { payload }: PayloadAction<Status>): Filter {
+      return { ...filter, status: payload };
+    },
+    setQuery(filter: Filter, { payload }: PayloadAction<string>): Filter {
```

Comment:
The reducer function `setQuery` should have the state as the first parameter and the action as the second parameter. Correct the function signature to: `setQuery(state: Filter, action: PayloadAction<string>): Filter { return { ...state, query: action.payload }; }`.